### PR TITLE
Simplified code for invocation of simple shell.

### PR DESCRIPTION
**Pull request by dgrr**
Also added HostKeyCallback parameter, password connection by default and term parameter now is determined by TERM env variable.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -112,17 +109,29 @@ func NewNativeConfig(user, clientVersion string, auth *Auth) (ssh.ClientConfig,
 		for _, p := range auth.Passwords {
 			authMethods = append(authMethods, ssh.Password(p))
 		}
+	} else {
+		pass := new(gopass.Shadow)
+
+		fmt.Printf("%s@%s's password: ", user, host)
+		err := pass.ReadPasswdMasked()
+		if err != nil {
+			panic(err)
```
Comment (glinton) [inline client.go]: Lets not panic

File: client.go [hunk 2]
```diff
@@ -70,26 +68,25 @@ type Auth struct {
 }
 
 // NewNativeClient creates a new Client using the golang ssh library
-func NewNativeClient(user, host, clientVersion string, port int, auth *Auth) (Client, error) {
-	if clientVersion == "" {
-		clientVersion = "SSH-2.0-Go"
+func NewNativeClient(user, hostname string, clientVersion []byte, auth *Auth) (Client, error) {
```
Comment (glinton) [inline client.go:73]: I'm not a fan of docker's go code for this reason. Lets create a new exported function with a new signature so we don't break things using this function

File: client.go [hunk 3]
```diff
@@ -70,26 +68,25 @@ type Auth struct {
...
-		Port:          port,
-		ClientVersion: clientVersion,
+		Hostname:      hostname,
+		ClientVersion: string(clientVersion),
 	}, nil
 }
 
 // NewNativeConfig returns a golang ssh client config struct for use by the NativeClient
-func NewNativeConfig(user, clientVersion string, auth *Auth) (ssh.ClientConfig, error) {
+func NewNativeConfig(user, host, clientVersion string, auth *Auth) (ssh.ClientConfig, error) {
```
Comment (glinton) [inline client.go:91]: See other comment about not modifying exported function signatures for compatibility. (If you've a better suggestion/reasoning, we're open to hear it)

File: client.go [hunk 4]
```diff
@@ -246,24 +264,28 @@ func (client *NativeClient) Shell(args ...string) error {
 		ssh.ECHO: 1,
 	}
 
-	fd := os.Stdin.Fd()
+	fd := int(os.Stdin.Fd())
 
-	if term.IsTerminal(fd) {
```
Comment (glinton) [inline client.go]: I want to say there was some reason we were using docker's `term` but I'll dig in and look at this again.

**Review by glinton — CHANGES_REQUESTED** [review]
In order to provide a clean library, lets return, or otherwise handle, the errors rather than panic. Also, I don't love the idea of changing function signatures but am open to suggestions.

File: client.go (hunk 1 above)
Comment (dgrr) [reply client.go]: Sorry. I wrote the panic temporarily to view the results.

File: client.go (hunk 4 above)
Comment (dgrr) [reply client.go]: Okay. I thought that golang library has support to windows terminal. But I'am wrong. I am going to restore term library.

Comment (glinton) [reply client.go]: It may have been added into the go lib since this was written, but if not term is preferred. Thanks

**Pull request closed without merging**